Support enabling service discovery for deployments

dfarrell07 opened the pull request:
See commit messages for details.

dfarrell07 committed:
Support lighthouse flag for E2E GHA and using=

Support `make e2e using=lighthouse` or setting a lighthouse variable in
Submariner's E2E GitHub Action.

Co-authored-by: Stephen Kitt <steve@sk2.org>
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
I added tests for this on my fork but they don't currently provide any useful feedback because the consuming project E2E always uses the Shipyard GHAs from the main branch, not the (potentially updated) local ones.

https://github.com/dfarrell07/shipyard/pull/6

dfarrell07 commented:
This is meant to fix the errors for Lighthouse E2E in submariner-operator caused by moving to Shipyard's E2E GHA, which didn't have a lighthouse param.

```
Unexpected input(s) 'lighthouse', valid inputs are ['cabledriver', 'deploytool', 'globalnet', 'working-directory']
```

https://github.com/submariner-io/submariner-operator/actions/runs/549914869

dfarrell07 commented:
Bump, we need to get this merged to actually test Lighthouse in submariner-operator.

skitt commented on Makefile.inc:
```diff
@@ -15,6 +15,10 @@ else ifneq (,$(filter wireguard,$(_using)))
 override DEPLOY_ARGS += --cable_driver wireguard
 endif
 
+ifneq (,$(filter lighthouse,$(_using)))
+override DEPLOY_ARGS += --deploytool_broker_args '--service-discovery'
+endif
```
Service discovery is enabled by default, don’t we need

```suggestion
ifneq (,$(filter lighthouse,$(_using)))
override DEPLOY_ARGS += --deploytool_broker_args '--service-discovery'
else
override DEPLOY_ARGS += --deploytool_broker_args '--service-discovery=false'
endif
```

instead?

dfarrell07 commented:
The Helm job failures are not related to this PR.

dfarrell07 commented:
@skitt Will you please re-review? I adopted your suggestion and added you as a co-author.

SteveMattar reviewed (changes requested):
The flags are not supported by `helm` charts deployment.
For example: instead of `--service-discovery=false` it should be `--set broker.serviceDiscovery=false`

my suggestion:
**deploy.sh**
![2021-03-04_04-11](https://user-images.githubusercontent.com/17129333/109900784-95840180-7ca0-11eb-8b42-1cb64c6fea5f.png)

**deploy-helm.sh**
![2021-03-04_05-17](https://user-images.githubusercontent.com/17129333/109906339-fc0d1d80-7ca8-11eb-9bd1-ab41bf15695d.png)


**deploy-operator.sh**
![2021-03-04_04-15](https://user-images.githubusercontent.com/17129333/109900849-ad5b8580-7ca0-11eb-9da6-1ce0c0109c20.png)

Then you can drop the `--deploytool_broker_args` and use this instead:

```sh
ifneq (,$(filter lighthouse,$(_using)))
override DEPLOY_ARGS +=  --serviceDiscovery
endif
```

mangelajo commented:
@dfarrell07 see @SteveMattar comments above  ^

dfarrell07 committed:
Support service discovery in deploy scripts

Add a flag to the deploy scripts to enable service discovery. Support
the flag in the Helm and operator deploy scripts.

Use the new deploy script flag in the using= flag.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Add lighthouse to deploy script CI matrix

Add the lighthouse service discovery flag to the deploy script CI
matrix.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>
